### PR TITLE
M24 rebase

### PR DESCRIPTION
**Pull request by jezdez**
Very few merge conflicts on this one.

@washort Mind taking a general look?

### CODE REVIEW

**Comment by washort** [comment]
Looks pretty clean - like you said, not significantly different from our previous tree. :+1:

**Comment by jezdez** [comment]
Sadly there seems to be some CI failures that look legit.

File: tests/factories.py [hunk 1]
```diff
@@ -176,6 +189,12 @@ def create_org(self, **kwargs):
 
         return org
 
+    def create_table_metadata(self, **kwargs):
+        return table_metadata_factory.create(**kwargs)
+
+    def create_column_metadata(self, **kwargs):
+        return column_metadata_factory.create(**kwargs)
+
```
Comment (washort) [inline tests/factories.py]: Here's a workaround that makes the tests pass:

```diff
@@ -286,7 +286,10 @@ class Factory(object):
             'org': self.org
         }
         args.update(kwargs)
-        return query_factory.create(**args)
+        q = query_factory.create(**args)
+        if 'schedule_failures' in args:
+            q.schedule_failures = args['schedule_failures']
+        return q
 
     def create_query_with_params(self, **kwargs):
         args = {
```

**Comment by washort** [comment]
Additionally, 194c80051 can be merged into 4684161e6

File: tests/factories.py (hunk 1 above)
Comment (jezdez) [reply tests/factories.py]: Ok, slapped it on 0b12ae3da9570412da3d9cf07f6c36f85ad93d24

**Pull request merged**